### PR TITLE
Feature/tri 370 implement endpoint get a list of jobs with processing state

### PR DESCRIPTION
**Pull request by ds-ext-ooyeneye**

### CODE REVIEW

File: irs-api/src/test/java/net/catenax/irs/controllers/IrsControllerTest.java [hunk 1]
```diff
@@ -76,13 +77,17 @@ void shouldReturnBadRequestWhenRegisterJobBodyNotValid(final RegisterJob registe
...
+        final JobStatusResult returnedJob = JobStatusResult.builder()
+                                                           .jobId(jobId.toString())
+                                                           .status(JobState.RUNNING.toString())
+                                                           .build();
+        when(service.getJobsByJobState(any())).thenReturn(List.of(returnedJob));
 
         this.mockMvc.perform(get("/irs/jobs"))
                     .andExpect(status().isOk())
-                    .andExpect(content().string(containsString(returnedJob.toString())));
+                    .andExpect(content().string(containsString(jobId.toString())));
```
Comment (ds-jkreutzfeld) [inline irs-api/src/test/java/net/catenax/irs/controllers/IrsControllerTest.java:90]: Wouldn't you also want to test for the correct status in the result object?

File: irs-api/src/test/java/net/catenax/irs/IrsApplicationTests.java [hunk 2]
```diff
@@ -55,10 +55,10 @@ void contextLoads() {
 
     @Test
     void generatedOpenApiMatchesContract() throws Exception {
-        final String generatedYaml =
-                this.restTemplate.getForObject("http://localhost:" + port + "/api/api-docs.yaml", String.class);
+        final String generatedYaml = this.restTemplate.getForObject("http://localhost:" + port + "/api/api-docs.yaml",
+                String.class);
         final String fixedYaml = Files.readString(new File("../api/irs-v1.0.yaml").toPath(), UTF_8);
-        assertThat(generatedYaml).isEqualToNormalizingNewlines(fixedYaml);
+        assertThat(generatedYaml).isEqualToIgnoringNewLines(fixedYaml);
```
Comment (ds-jkreutzfeld) [inline irs-api/src/test/java/net/catenax/irs/IrsApplicationTests.java:61]: Why this change? It used to work like this using isEqualToNormalizingNewlines

File: irs-api/src/main/java/net/catenax/irs/dto/JobStatusResult.java [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+package net.catenax.irs.dto;
+
+import lombok.Builder;
+import lombok.Data;
+import lombok.extern.jackson.Jacksonized;
+
+/**
+ * Response for job status request
+ */
+@Data
```
Comment (ds-jkreutzfeld) [inline irs-api/src/main/java/net/catenax/irs/dto/JobStatusResult.java:19]: Why not make this immutable with @Value?

**Review by ds-jkreutzfeld — CHANGES_REQUESTED** [review]

File: irs-api/src/main/java/net/catenax/irs/dto/JobStatusResult.java (hunk 3 above)
Comment (ds-ext-ooyeneye) [reply irs-api/src/main/java/net/catenax/irs/dto/JobStatusResult.java]: Done

File: irs-api/src/test/java/net/catenax/irs/controllers/IrsControllerTest.java (hunk 1 above)
Comment (ds-ext-ooyeneye) [reply irs-api/src/test/java/net/catenax/irs/controllers/IrsControllerTest.java]: Done

File: irs-api/src/main/java/net/catenax/irs/services/IrsItemGraphQueryService.java [hunk 4]
```diff
@@ -108,10 +108,16 @@ private JobParameter buildJobData(final @NonNull RegisterJob request) {
     }
 
     @Override
-    public List<UUID> getJobIdsByJobStates(final @NonNull List<JobState> jobStates) {
+    public List<JobStatusResult> getJobsByJobState(final @NonNull List<JobState> jobStates) {
         final List<MultiTransferJob> jobs = jobStore.findByStates(jobStates);
 
-        return jobs.stream().map(x -> x.getJob().getJobId()).collect(Collectors.toList());
+        return jobs.stream()
+                   .map(x -> JobStatusResult.builder()
```
Comment (ds-ext-kmassalski) [inline irs-api/src/main/java/net/catenax/irs/services/IrsItemGraphQueryService.java:115]: Rename x param to sth more sufficient

**Review by ds-ext-kmassalski — CHANGES_REQUESTED** [review]
```
[Jan Kreutzfeld] added a comment - [17/May/22 2:53]

- Modify response class to contain status as well as the jobId
- Fix default behavior to return jobs of all status instead of none (current behavior)
- Check whether pagination is implementable  (if easily done, then do it)
```
Only first bullet is implemented in the PR. You didnt change default implementation behaviour, what was expected by Martin. Also please leave comment about pagination possibilities.

File: irs-api/src/test/java/net/catenax/irs/IrsApplicationTests.java (hunk 2 above)
Comment (ds-jkreutzfeld) [reply irs-api/src/test/java/net/catenax/irs/IrsApplicationTests.java:61]: image
Maybe because you removed the newline at the end :)

**Comment by ds-ext-kmassalski** [comment]
@ds-ext-ooyeneye please answer or fix review comments!

**Comment by ds-ext-kmassalski** [comment]
Hi @ds-jhartmann @SaberIstos , can I ask for your opinion here?
Dapo and I had an open discussion this morning, if we should do requested changes for JobStatusResult class or not.

For me changing from String's to specific type properties (UUID, JobStates) gives:
- cleaner solution,
- easier to maintaine and underestand API model, 
- open–closed principle for further modifications/adjustments,
- no need to parse to string few times - spring's json parser does this anyway.

So the question is: what you think - should we stay with String, or do the change?

**Comment by ds-jhartmann** [comment]
Hi @ds-ext-kmassalski, to understand you correctly: You want to change 'String jobId' to 'UUID jobId' and 'String status' to 'JobState status'? Or do you have something different in mind?

**Comment by ds-ext-kmassalski** [comment]
"You want to change 'String jobId' to 'UUID jobId' and 'String status' to 'JobState status'?" - correct

**Comment by ds-jhartmann** [comment]
In this case I think we should do that. As you mentioned, it adds to maintainability, is cleaner to understand and provides precisely what the consumer of our API can expect as a result.

**Comment by SaberIstos** [comment]
After analyzing the changes in the above feature branch, I guess that using the benefit of the spring json parser is the common way instead of parsing the jobId as a string for several times. So the usage of UUID as a data type is better from my point of view.

File: irs-api/src/test/java/net/catenax/irs/IrsApplicationTests.java (hunk 2 above)
Comment (ds-ext-ooyeneye) [reply irs-api/src/test/java/net/catenax/irs/IrsApplicationTests.java:61]: Resolved

**Review by ds-ext-kmassalski — CHANGES_REQUESTED** [review]
Add test to assert status and bussing logic (findAdd vs findByStates)

**Review by ds-ext-kmassalski — APPROVED** [review]

**Review by ds-ext-kmassalski — APPROVED** [review]

**Pull request merged**